crypto/rsa/rsa_pss.c

```diff
@@ -191,6 +201,8 @@ int RSA_padding_add_PKCS1_PSS_mgf1(RSA *rsa, unsigned char *EM,
         if (RAND_bytes(salt, sLen) <= 0)
             goto err;
     }
+    if (hLen < 1 || hLen - 1 > emLen)
```

Comment:
so you want to predict if "emLen - hLen - 1" will be negative.
but algebically this is transforms like follows:
emLen - hLen - 1 < 0
emLen < hLen + 1

IOW `hLen + 1 > emLen`